code/src/sixsq/nuvla/server/resources/session.clj

```diff
@@ -354,24 +354,30 @@ status, a 'set-cookie' header, and a 'location' header with the created
 (def edit-impl (std-crud/edit-fn resource-type))
 
 (defn update-cookie-session
-  [{:keys [id user roles active-claim client-ip] :as session}
-   {headers :headers {:keys [claim]} :body :as request}]
-  (let [updated-claims (-> roles
-                           authn-info/split-claims
-                           (disj (or active-claim user))
-                           (conj claim))
-        updated-roles  (->> updated-claims sort (str/join " "))
+  [{:keys [id user user-groups client-ip] :as session}
+   {headers :headers {:keys [claim extended]} :body :as request}]
+  (let [extended-claim (when extended
+                         (->> user-groups
+                              :subgroups
+                              (filter #((set (:parents %)) claim))
```

Comment:
Could be (comp claim set :parents) to avoid #()

Reply:
claim is a string